File: code_samples/kotlin/testing_unit_tests.kt
/**
 * Testing: Unit Tests
 * AI/ML Training Sample
 */

class UnitTests {
    var data: String = ""
        private set
    
    fun process(input: String) {
        data = input
    }
    
    fun getData(): String = data
    
    fun validate(): Boolean = data.isNotEmpty()
}

fun main() {
    val instance = UnitTests()
    instance.process("example")
    println("Data: ${instance.getData()}")
    println("Valid: ${instance.validate()}")
}

File: code_samples/dart/testing_unit_tests.dart
// Testing: Unit Tests
// AI/ML Training Sample

class UnitTests {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = UnitTests();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
